### PR TITLE
Add jpeg2000 support

### PR DESCRIPTION
**Pull request by youri-k**
### Description:
- adds `jp2` support by including it in supported file endings
- fixes the type translation for `signed int` layer

### Todos:
 - [X] Updated Changelog

### CODE REVIEW

File: wkcuber/wkcuber/image_readers.py [hunk 1]
```diff
@@ -448,6 +448,7 @@ def __init__(self) -> None:
             ".tiff": TiffImageReader(),
             ".jpg": PillowImageReader(),
             ".jpeg": PillowImageReader(),
+            ".jp2": PillowImageReader(),
```
Comment (jstriebel) [inline wkcuber/wkcuber/image_readers.py:451]: https://en.wikipedia.org/wiki/JPEG_2000 suggests more possible file-endings:
.jp2, .j2k, .jpf, .jpm, .jpg2, .j2c, .jpc, .jpx, .mj2

Should we add them all?

Comment (youri-k) [reply wkcuber/wkcuber/image_readers.py:451]: Sounds good. I'll add them.

Comment (youri-k) [reply wkcuber/wkcuber/image_readers.py:457]: I just checked whether there are more file endings for JPEG as well, so I added `.jpe, .jfif, .jif` as suggested here https://en.wikipedia.org/wiki/JPEG#JPEG_filename_extensions

**Review by jstriebel — APPROVED** [review]
LGTM :+1:

**Pull request merged**